packages/loopring_v3/circuit/Utils/Data.h

```diff
@@ -222,353 +340,371 @@ class Ring
-    block.protocolMakerFeeBips = ethsnarks::FieldT(j["protocolMakerFeeBips"].get<unsigned int>());
-
-    block.signature = j.at("signature").get<Signature>();
-
-    block.accountUpdate_P = j.at("accountUpdate_P").get<AccountUpdate>();
-
-    block.operatorAccountID = ethsnarks::FieldT(j.at("operatorAccountID"));
-    block.accountUpdate_O = j.at("accountUpdate_O").get<AccountUpdate>();
-
-    // Read settlements
-    json jRingSettlements = j["ringSettlements"];
-    for(unsigned int i = 0; i < jRingSettlements.size(); i++)
-    {
-        block.ringSettlements.emplace_back(jRingSettlements[i].get<Loopring::RingSettlement>());
-    }
+    withdrawal.accountID = ethsnarks::FieldT(j.at("accountID"));
+    withdrawal.tokenID = ethsnarks::FieldT(j.at("tokenID"));
+    withdrawal.amount = ethsnarks::FieldT(j["amount"].get<std::string>().c_str());
+    withdrawal.feeTokenID = ethsnarks::FieldT(j.at("feeTokenID"));
+    withdrawal.fee = ethsnarks::FieldT(j["fee"].get<std::string>().c_str());
+    withdrawal.to = ethsnarks::FieldT(j["to"].get<std::string>().c_str());
+    withdrawal.dataHash = ethsnarks::FieldT(j["dataHash"].get<std::string>().c_str());
+    withdrawal.minGas = ethsnarks::FieldT(j.at("minGas"));
+    withdrawal.type = ethsnarks::FieldT(j.at("type"));
 }
 
-class Deposit
+
+class AccountUpdateTx
 {
 public:
-    ethsnarks::FieldT amount;
-    BalanceUpdate balanceUpdate;
-    AccountUpdate accountUpdate;
+    ethsnarks::FieldT accountID;
+    ethsnarks::FieldT publicKeyX;
+    ethsnarks::FieldT publicKeyY;
+    ethsnarks::FieldT walletHash;
+    ethsnarks::FieldT feeTokenID;
+    ethsnarks::FieldT fee;
+    ethsnarks::FieldT type;
 };
 
-static void from_json(const json& j, Deposit& deposit)
+static void from_json(const json& j, AccountUpdateTx& update)
 {
-    deposit.amount = ethsnarks::FieldT(j.at("amount").get<std::string>().c_str());
-    deposit.balanceUpdate = j.at("balanceUpdate").get<BalanceUpdate>();
-    deposit.accountUpdate = j.at("accountUpdate").get<AccountUpdate>();
+    update.accountID = ethsnarks::FieldT(j.at("accountID"));
+    update.publicKeyX = ethsnarks::FieldT(j["publicKeyX"].get<std::string>().c_str());
+    update.publicKeyY = ethsnarks::FieldT(j["publicKeyY"].get<std::string>().c_str());
+    update.walletHash = ethsnarks::FieldT(j["walletHash"].get<std::string>().c_str());
+    update.feeTokenID = ethsnarks::FieldT(j.at("feeTokenID"));
+    update.fee = ethsnarks::FieldT(j["fee"].get<std::string>().c_str());
+    update.type = ethsnarks::FieldT(j.at("type"));
 }
 
-class DepositBlock
+
+class NewAccount
 {
 public:
-    ethsnarks::FieldT exchangeID;
-
-    ethsnarks::FieldT merkleRootBefore;
-    ethsnarks::FieldT merkleRootAfter;
-
-    ethsnarks::LimbT startHash;
-
-    ethsnarks::FieldT startIndex;
-    ethsnarks::FieldT count;
-
-    std::vector<Loopring::Deposit> deposits;
+    ethsnarks::FieldT payerAccountID;
+    ethsnarks::FieldT feeTokenID;
+    ethsnarks::FieldT fee;
```

Comment:
I suggest to remove the "new" prefix and try to keep the field in the same order as in AccountUpdate